backend/routes/rides.js

```diff
@@ -16,20 +17,29 @@ router.get('/', async function (req, res, next) {
     //TODO: validate login state - login user = given userid
     //TODO: search logic for begin and end date
     const query = JSON.parse(req.query['query']);
-    console.log(query);
+    if (DEBUG) {
+        console.log(query);
+    }
     const originCoords = query['originCoords'];
     const destCoords = query['destCoords'];
-    const beginDate = query['beginDate'];
-    const endDate = query['endDate'];
+    // const beginDate = query['beginDate'];
+    // const endDate = query['endDate'];
+    const time = query['time'];
+    // build date range
+    const rideTime = new Date(time);
+    const dayStart = new Date(rideTime.getFullYear(), rideTime.getMonth(), rideTime.getDate());
+    const dayEnd = new Date(rideTime.getFullYear(), rideTime.getMonth(), rideTime.getDate() + 1);
+
     const distance = query['distance']; // within x miles
     const collection = client.dbCollection(collectionName);
     // const METERS_PER_MILE = 1609.34;
     const distInRadians = distance / 3963.2;//converts the distance to radians by dividing by the approximate equatorial radius of the earth
 
     collection.find({
-        time: {$gte: beginDate, $lte: endDate},
-        originCoords: {$geoWithin: {$centerSphere: [originCoords, distInRadians]}},
-        destCoords: {$geoWithin: {$centerSphere: [destCoords, distInRadians]}}
+        // time: {$gte: beginDate, $lte: endDate},
+        time: {$gte: dayStart, $lte: dayEnd},
+        // originCoords: {$geoWithin: {$centerSphere: [originCoords, distInRadians]}},
```

Comment:
Instead of disabling the original search with coords, a better way of doing it would be
```
var filter;
if !originCoords 
    filter = {time: {$gte: xx, xxxx}};
else
    filter = {time: {$gte: xx, xxxx}, originCoords: {}, destCoords: {}}

collection.find(filter).toArray(xxx);

```
So this way you can support different types of search query at the same time.
Or let me know what kind of customized search queries there will be, so I can redesign this function to make it more general.

Reply:
Good thinking, I'll make these changes